Code/Engine/RendererFoundation/Profiling/Implementation/Profiling.cpp

```diff
@@ -30,13 +30,18 @@ class GPUProfilingSystem
     {
       auto& timingScope = s_TimingScopes.PeekFront();
 
+      ezTime beginTime;
       ezTime endTime;
-      if (e.m_pDevice->GetTimestampResult(timingScope.m_EndTimestamp, endTime).Succeeded())
+      ezEnum<ezGALAsyncResult> resBegin = e.m_pDevice->GetTimestampResult(timingScope.m_BeginTimestamp, beginTime);
+      ezEnum<ezGALAsyncResult> resEnd = e.m_pDevice->GetTimestampResult(timingScope.m_EndTimestamp, endTime);
+
+      if (resBegin == ezGALAsyncResult::Expired || resEnd == ezGALAsyncResult::Expired)
       {
-        ezTime beginTime;
-        EZ_VERIFY(e.m_pDevice->GetTimestampResult(timingScope.m_BeginTimestamp, beginTime).Succeeded(),
-          "Begin timestamp should be finished before end timestamp");
+        s_TimingScopes.PopFront();
+      }
 
+      if (resBegin == ezGALAsyncResult::Ready || resEnd == ezGALAsyncResult::Ready)
```

Comment:
Shouldn't that be resBegin == ezGALAsyncResult::Ready && resEnd == ezGALAsyncResult::Ready

Reply:
Yes totally!